### DISCUSSION TITLE
CapRover (fka Captain Duck Duck)

### DISCUSSION THREAD

**Discussion by michaelkremmel** [crate/crate-clients-tools#195] [Ideas]
CapRover presents itself as free and open source platform as a service and
> an extremely easy to use app/database deployment & web server manager for your NodeJS, Python, PHP, ASP.NET, Ruby, MySQL, MongoDB, Postgres, WordPress (and etc...) applications!
> 
> It's blazingly fast and very robust as it uses Docker, nginx, LetsEncrypt and NetData under the hood behind its simple-to-use interface.

It supports many databases (such as ArangoDB, CouchBase, InfluxDB, MariaDB, MongoDB, MySQL, PostgreSQL etc) as one click app. 

CrateDB is so far not part of it, but could be added [here ](https://github.com/caprover/one-click-apps). So my suggestion would be to have a look at it if would make sense to add CrateDB there because I saw that CapRover was running on a prospect's server who was trying out CrateDB.

**Comment by amotl** [comment]
Hi @michaelkremmel,

thanks a stack for your suggestion.

> CapRover supports many databases (such as ArangoDB, CouchBase, InfluxDB, MariaDB, MongoDB, MySQL, PostgreSQL etc) as one click app.

Indeed, many databases are [listed on CapRover's registry](https://github.com/caprover/one-click-apps/tree/master/public/v4/apps), like [PostgreSQL](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/postgres.yml), [InfluxDB 1.x](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/influxdb.yml), [InfluxDB 2.x](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/influxdb2.yml), [Elasticsearch](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/elasticsearch.yml), [MongoDB](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/mongodb.yml), and many more. Excellent.

> CrateDB is so far not part of it, but could be added [here](https://github.com/caprover/one-click-apps).

In the spirit that all others are partying already, it is clear that CrateDB is missing again, and should certainly be added in one way or another. Thanks.

In most of the cases, this would roughly be deriving from the [PostgreSQL](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/postgres.yml) adapter in a good way. In this case, it looks like it is applicable to copy that `postgres.yml` into a `cratedb.yml`, including relevant adjustments for CrateDB?

Do you think "that's it"? I didn't do many investigations into CapRover yet, so I certainly would need additonal education to be able to share any serious evaluation here. I am just picking up that ticket and thought it would be a good idea to probe it.

With kind regards,
Andreas.
